cli/azd/pkg/tools/bicep/bicep.go

```diff
@@ -93,47 +89,22 @@ func (cli *bicepCli) CheckInstalled(ctx context.Context) (bool, error) {
 }
 
 func (cli *bicepCli) Build(ctx context.Context, file string) (string, error) {
-	sniffCliVersion := func() (string, error) {
-		verRes, err := cli.runCommand(ctx, "version", "--out", "json")
-		if err != nil {
-			return "", fmt.Errorf("failing running az version: %s (%w)", verRes.String(), err)
-		}
-
-		var jsonVer struct {
-			AzureCli string `json:"azure-cli"`
-		}
-
-		if err := json.Unmarshal([]byte(verRes.Stdout), &jsonVer); err != nil {
-			return "", fmt.Errorf("parsing cli version json: %s: %w", verRes.Stdout, err)
-		}
-
-		return jsonVer.AzureCli, nil
-	}
-
-	args := []string{"bicep", "build", "--file", file, "--stdout"}
-
-	// Workaround azure/azure-cli#22621, by passing `--no-restore` to the CLI when
-	// when version 2.37.0 is installed.
-	if ver, err := sniffCliVersion(); err != nil {
-		log.Printf("error sniffing az cli version: %s", err.Error())
-	} else if ver == "2.37.0" {
-		log.Println("appending `--no-restore` to bicep arguments to work around azure/azure-dev#22621")
-		args = append(args, "--no-restore")
-	}
```

Comment:
@ellismg Do we still need this check? If I recall the history of this issue the issue was with the `az cli` bicep wrapper and not bicep itself.

Reply:
Oh, so sorry Walace - I was waiting for a CI job to run and ended up opening #1074 a few hours before you.  You are correct that this code is no longer needed, the bug was with the wrapper itself.